api/lifetime_watcher.go

```diff
@@ -268,18 +268,20 @@ func (r *LifetimeWatcher) doRenew() error {
 		default:
 		}
 
-		var leaseDuration time.Duration
+		var remainingLeaseDuration time.Duration
 		fallbackLeaseDuration := initialTime.Add(priorDuration).Sub(time.Now())
```

Comment:
One last thought, I think the value of this only gets used in one place now, so you could delete the variable to simplify a bit.